packages/taco-auth/src/storage.ts

```diff
@@ -36,11 +38,24 @@ export class LocalStorage {
         : new BrowserStorage();
   }
 
-  getItem(key: string): string | null {
-    return this.storage.getItem(key);
+  public getAuthSignature(key: string): AuthSignature | null {
+    const asJson = this.storage.getItem(key);
+    if (!asJson) {
+      return null;
+    }
+    return LocalStorage.fromJson(asJson);
+  }
+
+  public setAuthSignature(key: string, authSignature: AuthSignature): void {
+    const asJson = LocalStorage.toJson(authSignature);
+    this.storage.setItem(key, asJson);
+  }
+
+  public static fromJson(json: string): AuthSignature {
+    return authSignatureSchema.parse(JSON.parse(json));
   }
 
-  setItem(key: string, value: string): void {
-    this.storage.setItem(key, value);
+  public static toJson(signature: AuthSignature): string {
```

Comment:
Not a blocker but it's a bit weird that json<=>authsignature conversion methods are part of the storage interface